### DISCUSSION TITLE
Fail get clone package from private github repository

### DISCUSSION THREAD

**Discussion by ekokurniadi** [codemagic-ci-cd/codemagic-docs#2112] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Hello, I have a Windows desktop flutter project that uses packages taken from a private GitHub repository, but it fails when doing Flutter Pub Get in Codemagic because of SSH problems, I have added the SSH key to the Codemagic environment but it still fails, can you help me?

### Expected results

Can clone private packages into the project using the ssh key that has been added

### Actual results

Project failed to build because SSH was not read

**Comment by himesh-cm** [comment]
Hi @ekokurniadi, I am unable to reproduce the issue you reported. Can you please provide me with your build ID so that I can investigate the issue further?

**Reply by ekokurniadi** [reply]
Hi @himesh-cm, thank's for your response. 
Sure, this my Build ID and link for the build app 
Build ID : 652ea448ccb85431f9011884
Link ID : https://codemagic.io/app/640058ca7710614a93e38aa6/build/652ea448ccb85431f9011884

**Reply by stngcoding** [reply]
You have to put your ssh key in global variables 
after that add this to your codemagic.yaml at the first of "script" section

`- name: Script that uses an alternative SSH key
        script: |
          #!/usr/bin/env bash
          echo "${MODULE_SSH_KEY}" > /tmp/ssh_key
          chmod 600 /tmp/ssh_key
          eval `ssh-agent -s`
          ssh-add /tmp/ssh_key`

**Reply by ekokurniadi** [reply]
i have multiple virtual machine,  can i use single ssh for windows, and mac ?

**Reply by stngcoding** [reply]
sure but each time you do pub install the private repo will requires SHA256 key in your 'SSH and GPG keys'
unless those VMs uses the same git account

**Reply by himesh-cm** [reply]
Hi @ekokurniadi, Please note that you should contact our team through the chat widget in the UI for a quicker response, as your team is on a paid plan.

Regarding your query, 
You need to generate an `SSH` key pair to access a private repo. You will find all the details related to it in our docs [here](https://docs.codemagic.io/knowledge-git/access-private-git-submodules/).

Let me know, if you run into any issues.

**Comment by himesh-cm** [comment]
Hi @stngcoding, I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
